src/main/java/com/amazon/ion/impl/macro/EncodingContext.kt

```diff
@@ -6,13 +6,19 @@ package com.amazon.ion.impl.macro
  * When we implement modules, this will likely need to be replaced.
  * For now, it is a placeholder for what is to come and a container for the macro table.
  */
-class EncodingContext(macroTable: Map<MacroRef, Macro>) {
+class EncodingContext {
 
-    val macroTable = macroTable.toMutableMap()
+    val macroTable: MacroTable
```

Comment:
The extra layer of indirection is to allow us to use SystemMacro as the effective MacroTable when we don't have a different macro table defined- always reading through to the same constant macro table. The alternative as I see it  is to special-case "have we dropped the system module?" in various places rather than keeping a consistent abstraction. 

... oh, I just realized that you might mean "the EncodingContext" as the extra layer of indirection rather than "the existence of a MacroTable" as the extra layer of indirection. Which is it?

Reply:
The existence of both `EncodingDirective` and `MacroTable` makes it seem like one of them is an extra layer of indirection when we could just have something like this:
```kotlin
class EncodingContext(
    val symbolTable: Array<String?>,
    val macroTable: Map<MacroRef, Macro>,
    val activeModules: Map<String, Module>,
    val antecedentSymbolTable: Array<String?> = emptyArray(),
    val antecedentMacroTable: Map<MacroRef, Macro> = emptyMap(),
)
```
(With or without antecedent chaining.)